gnomad_qc/v4/create_release/validate_and_export_vcf.py

```diff
@@ -1232,22 +1292,44 @@ def main(args):
             ht = res.release_ht.ht()
             validated_ht = res.validated_ht.ht()
 
-            # v4 Genomes drops subsets from VCF
-            subsets = SUBSETS["exomes"] if data_type == "exomes" else []
-
-            header_dict = prepare_vcf_header_dict(
-                ht,
-                validated_ht,
-                bin_edges=make_hist_bin_edges_expr(
+            if not for_joint:
+                # v4 Genomes drops subsets from VCF
+                subsets = SUBSETS["exomes"] if data_type == "exomes" else []
+                header_dict = prepare_vcf_header_dict(
                     ht,
-                    include_age_hists=True,
-                ),
-                age_hist_distribution=hl.eval(ht.age_distribution.bin_freq),
-                subset_list=subsets,
-                pops=POPS[data_type],
-                data_type=data_type,
-                joint_included=joint_included,
-            )
+                    validated_ht=validated_ht,
+                    info_fields=list(validated_ht.info),
+                    bin_edges=make_hist_bin_edges_expr(
+                        ht,
+                        include_age_hists=True,
+                    ),
+                    age_hist_distribution=hl.eval(ht.age_distribution.bin_freq),
+                    subset_list=subsets,
+                    pops=POPS[data_type],
+                    data_type=data_type,
+                    joint_included=joint_included,
+                )
+            else:
+                header_dict = {"info": {}}
+                for dt in ["exomes", "genomes", "joint"]:
+                    dt_ht = select_type_from_joint_ht(ht, dt)
+                    temp_header_dict = prepare_vcf_header_dict(
+                        dt_ht,
+                        validated_ht=validated_ht,
+                        info_fields=[f for f in validated_ht.info.keys() if dt in f],
+                        bin_edges=make_hist_bin_edges_expr(
+                            dt_ht, include_age_hists=True
+                        ),
+                        age_hist_distribution=hl.eval(dt_ht.age_distribution.bin_freq),
+                        subset_list=[dt],
```

Comment:
The way this runs, we add INFO fields that we do not have in the joint VCF, e.g. `AS_pab_max`, `AS_SB_TABLE`,  because of the call to `add_as_info_dict` which is made when `subset_list = ['exomes']` or `subset_list = ['genomes']` . The built in logic on L867 only accounts for when joint is the subset and data type. We don't want those fields as they are not in the joint HT.

Reply:
I had the older logical which was a bit redundant, so I changed it in my new commit. Also, I forgot to implement the merged make_filter_info_dict in this PR, but in the export_vcf PR, I have a better position to put that one.